src/services/RedocNormalizedOptions.ts

```diff
@@ -153,6 +155,20 @@ export class RedocNormalizedOptions {
     return 0;
   }
 
+  static normalizeDownloadFileName(value: RedocRawOptions['downloadFileName']): string {
+    if (value) {
+      const extname = path.extname(value);
+      if (extname === '.json' || extname === '.yaml') {
+        return value;
+      } else {
+        console.warn(`downloadFileName must be a JSON or YAML file.`);
+      }
+    }
+
+    // Default value
+    return 'swagger.json';
```

Comment:
I think you should check the version of the definition (to keep backward compatibility) or return the default name `openapi.yaml`.

Reply:
```suggestion
    return 'openapi.json';
```